src/main/java/entities/Room.java

```diff
@@ -0,0 +1,159 @@
+                add(topic.getRole());  // allow this topics' mentors
+                add(mentee);  // allow the mentee
+            }
+        };
+        setChannelPermissions(textChannel, guild.getPublicRole(), allowList);
+
+        voiceChannel = category.createVoiceChannel(name).complete();
+        setChannelPermissions(voiceChannel, guild.getPublicRole(), allowList);
+    }
+
+    /**
+     * Create a Room name from a Topic name and room number.
+     * @param topicName The name of the owning Topic
+     * @param roomNumber The room's number
+     * @return The Room's name
+     */
+    public static String makeName(String topicName, int roomNumber) {
+        return String.format("%s-%d", topicName, roomNumber);
+    }
+
+    /**
+     * Deny view permissions to `veryoneRole` and allow view permissions to all
+     * roles/members in `allowList`.
+     * @param channel The channel to apply permission overrides to 
+     * @param everyoneRole Reference to `@everyone` role
+     * @param allowList List of roles/members that should have access to this
+     *     channel. The bot's user MUST be in this list.
+     */
+    private void setChannelPermissions(GuildChannel channel, IPermissionHolder everyoneRole, Collection<IPermissionHolder> allowList) {
+        for (IPermissionHolder holder : allowList) {
+            channel.putPermissionOverride(holder)
+                .setAllow(Permission.VIEW_CHANNEL)
+                .complete();
+        }
+
+        channel.putPermissionOverride(everyoneRole)
+            .setDeny(Permission.VIEW_CHANNEL)
+            .complete();
+    }
+
+    /**
+     * Delete this room's channels from the server. This object should be disposed
+     * of after calling this method.
+     */
+    public void delete() {
+        textChannel.delete().complete();
+        voiceChannel.delete().complete();
+    }
+
+    /**
+     * Delete any existing channels using this name/number
+     */
+    public void deleteExisting() {
+        Optional<TextChannel> optionalTextChannel = category.getTextChannels().stream()
+            .filter(tc -> tc.getName().equals(name.toLowerCase()))  // text channels are lowercase
+            .findFirst();
+        optionalTextChannel.ifPresent(tc -> tc.delete().complete());
+
+        Optional<VoiceChannel> optionalVoiceChannel = category.getVoiceChannels().stream()
+            .filter(vc -> vc.getName().equals(name))
+            .findFirst();
+        optionalVoiceChannel.ifPresent(vc -> vc.delete().complete());
+    }
+
+    /**
+     * Create an invite to this room's voice channel. This invite can be used
+     * twice and expires after 5 minutes.
+     * @return A new Invite for this room's voice channel
+     */
+    public Invite getVoiceChannelInvite() {
+        InviteAction action = voiceChannel.createInvite();
+        action.setMaxAge(5 * 60);  // 5 minutes, to prevent hitting the invite cap
+        action.setMaxUses(2);
```

Comment:
That can only happen if another mentor does it, since regular users don't have permission to enter that channel. However, your point does hold up if another mentor accidentally clicks it. I'll bump it up to 5 max uses; with the 5 minute timer, we probably won't hit our invite limit either way.